src/matrix_cookbook/3_inverses.lean

```diff
@@ -87,13 +89,79 @@ end
 
 /-! ### The Searle Set of Identities -/
 
-lemma eq_161 : (1 + A⁻¹)⁻¹ = A⬝(A + 1)⁻¹ := sorry
-lemma eq_162 : (A + Bᵀ⬝B)⁻¹⬝B = A⁻¹⬝B⬝(1 + Bᵀ⬝A⁻¹⬝B)⁻¹:= sorry
-lemma eq_163 : (A⁻¹ + B⁻¹)⁻¹ = A⬝(A + B)⁻¹⬝B ∧ (A⁻¹ + B⁻¹)⁻¹ = B⬝(A + B)⁻¹⬝A := sorry
-lemma eq_164 : A - A⬝(A + B)⁻¹⬝A = B - B⬝(A + B)⁻¹⬝B := sorry
-lemma eq_165 : A⁻¹ + B⁻¹ = A⁻¹⬝(A + B)⁻¹⬝B⁻¹ := sorry
-lemma eq_166 : (1 + A⬝B)⁻¹ = 1 - A⬝(1 + B⬝A)⁻¹⬝B := sorry
-lemma eq_167 : (1 + A⬝B)⁻¹⬝A = A⬝(1 + B⬝A)⁻¹ := sorry
```

Comment:
I think 161, 163, 164, and 165 are true more generally for `ring.inverse` and `is_unit A`; on arbitrary rings, not just on matrices. Can you prove those versions in a for-mathlib file?

Reply:
I am not sure what instance type to use for `ring.inverse` but I used a `division_ring` with `group` to have associativity